addons/mail/controllers/discuss.py

```diff
@@ -432,6 +432,20 @@ def mail_thread_data(self, thread_model, thread_id, request_list, **kwargs):
         thread = request.env[thread_model].with_context(active_test=False).search([('id', '=', thread_id)])
         return thread._get_mail_thread_data(request_list)
 
+    def _mail_thread_messages_filter(self):
+        """ Allows to add AND clause on mail.message query to filter further the message returned in the chatter.
```

Comment:
> Allows to add AND clause on mail.message query

Technically you can modify the domain as you want (not just with AND clause)

Reply:
I have reworded it as "Allows to refine mail.message query..." to make it more generic but actually the methods append condition in a "AND" block. That was what I wanted to say but it was not clear enough.